github/repos.go

```diff
@@ -557,6 +557,12 @@ type RequiredStatusChecks struct {
 	Contexts []string `json:"contexts"`
 }
 
+// RequiredStatusChecksRequest represents a request to edit a protected branch's status checks.
+type RequiredStatusChecksRequest struct {
+	Strict   *bool     `json:"strict"`
+	Contexts *[]string `json:"contexts"`
```

Comment:
A slice is already a reference object, so this should be `[]string` and not `*[]string`.
Also, please add `omitempty` to this one as well.

Reply:
Will do for `omitempty`.

How important is consistency? I was trying to match the other code already in the repo, and other request objects with optional slices use pointers -- for example, in the same file:

```
type DismissalRestrictionsRequest struct {
	// The list of user logins who can dismiss pull request reviews. (Required; use nil to disable dismissal_restrictions or &[]string{} otherwise.)
	Users *[]string `json:"users,omitempty"`
	// The list of team slugs which can dismiss pull request reviews. (Required; use nil to disable dismissal_restrictions or &[]string{} otherwise.)
	Teams *[]string `json:"teams,omitempty"`
}
```

Reply:
Actually on second glance it looks like the file has both styles:

```go
type TransferRequest struct {
	NewOwner string  `json:"new_owner"`
	TeamID   []int64 `json:"team_id,omitempty"`
}
```

Will remove pointer reference.

Reply:
Consistency is very important in this repo, but in this case, the `*[]string` is not required I don't believe because an empty list here would just mean no contexts (if I'm reading the docs right). Do you agree?

Reply:
Yes, agree and updated in latest PR update.